File: app/frontend/store/entities/hub-meta-store.js
// Wire protocol v2 — singleton entity stores for hub metadata.
//
// Both `hub` (lifecycle / recovery state) and `connection_code` (pairing
// QR + URL) are modelled as singleton entity types whose id is the hub_id
// (design brief §3 + §12.7). Each store typically holds exactly one entity.

import { createEntityStore } from './createEntityStore'

export const useHubMetaStore = createEntityStore('hub', { idField: 'hub_id' })

export const useConnectionCodeStore = createEntityStore('connection_code', {
  idField: 'hub_id',
})

File: cli/src/clients/tui/entity_stores/hub_meta_store.rs
//! Hub-singleton entity stores. Both `hub` (recovery state) and
//! `connection_code` are modelled as singleton entity types whose id is
//! the hub_id, per design brief §3 + §12.7. Each store typically holds
//! exactly one entity.

use super::EntityStore;

/// Backing store for the singleton `hub` entity type (lifecycle / recovery
/// state). Id field is `hub_id`.
pub type HubMetaStore = EntityStore;

/// Backing store for the singleton `connection_code` entity type (pairing
/// QR + URL). Id field is `hub_id`.
pub type ConnectionCodeStore = EntityStore;
